skyrim-platform/src/platform_se/skyrim_platform/GameEventSinks.cpp

```diff
@@ -665,3 +732,1297 @@ RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::TESTriggerEnterEvent* event,
+  RE::BSTEventSource<RE::TESTriggerEnterEvent>* eventSource)
+{
+  auto converted =
+    reinterpret_cast<const TESEvents::TESTriggerEnterEvent*>(event);
+
+  if (!converted) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto target = converted->target.get();
+  auto cause = converted->cause.get();
+  auto targetId = target ? target->formID : 0;
+  auto causeId = cause ? cause->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [target, cause, targetId, causeId] {
+      auto obj = JsValue::Object();
+
+      auto targetLocal = RE::TESForm::LookupByID(targetId);
+      auto causeLocal = RE::TESForm::LookupByID(causeId);
+
+      if (targetLocal == nullptr || causeLocal == nullptr ||
+          targetLocal != target || causeLocal != cause) {
+        return;
+      }
+
+      obj.SetProperty("target", CreateObject("ObjectReference", targetLocal));
+      obj.SetProperty("cause", CreateObject("ObjectReference", causeLocal));
+
+      EventsApi::SendEvent("triggerEnter", { JsValue::Undefined(), obj });
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::TESTriggerLeaveEvent* event,
+  RE::BSTEventSource<RE::TESTriggerLeaveEvent>* eventSource)
+{
+  auto converted =
+    reinterpret_cast<const TESEvents::TESTriggerLeaveEvent*>(event);
+
+  if (!converted) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto target = converted->target.get();
+  auto cause = converted->cause.get();
+  auto targetId = target ? target->formID : 0;
+  auto causeId = cause ? cause->formID : 0;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask(
+    [target, cause, targetId, causeId] {
+      auto obj = JsValue::Object();
+
+      auto targetLocal = RE::TESForm::LookupByID(targetId);
+      auto causeLocal = RE::TESForm::LookupByID(causeId);
+
+      if (targetLocal == nullptr || causeLocal == nullptr ||
+          targetLocal != target || causeLocal != cause) {
+        return;
+      }
+
+      obj.SetProperty("target", CreateObject("ObjectReference", targetLocal));
+      obj.SetProperty("cause", CreateObject("ObjectReference", causeLocal));
+
+      EventsApi::SendEvent("triggerLeave", { JsValue::Undefined(), obj });
+    });
+
+  return RE::BSEventNotifyControl::kContinue;
+}
+
+RE::BSEventNotifyControl GameEventSinks::ProcessEvent(
+  const RE::TESSleepStartEvent* event,
+  RE::BSTEventSource<RE::TESSleepStartEvent>* eventSource)
+{
+  auto converted =
+    reinterpret_cast<const TESEvents::TESSleepStartEvent*>(event);
+
+  if (!converted) {
+    return RE::BSEventNotifyControl::kContinue;
+  }
+
+  auto sleepStart = converted->sleepStartTime;
+  auto sleepEnd = converted->desiredSleepEndTime;
+
+  SkyrimPlatform::GetSingleton().AddUpdateTask([sleepStart, sleepEnd] {
+    auto obj = JsValue::Object();
+
+    if (sleepStart < 0 || sleepEnd < 0) {
```

Comment:
What's the meaning of this condition? Seems to be always false.
Please add a comment if this check is needed or remove.